docs: add changelog and complete documentation (#13)
eduNEXT/tutor-contrib-codejail: README.rst

> codejail plugin for `Tutor <https://docs.tutor.overhang.io>`__

Reviewer: Please, add the type of language to the bash instructions!

===================================================================================

> The repository Tutor contrib codejail, is a codejail plugin to manages execution of
> untrusted code in secure sandboxes. It is designed primarily for Python execution, but
> can be used for other languages as well. Security is enforced with AppArmor.

Reviewer: ```suggestion
Tutor plugin that enables execution of untrusted code in secure sandboxes using an external [service](https://github.com/eduNEXT/codejailservice) based on the [codejail](https://github.com/openedx/codejail) library.
```
Most of what you described here is part of codejail itself, and this is just a tutor plugin that helps with tutor + codejail.

    tutor local quickstart

Compatibility
-------------

+-------------------+----------------+
| Open edX Release  |  Tutor Version |
+===================+================+
+-------------------+----------------+
|       Lilac       |    >= 12.x     |
+-------------------+----------------+
|       Maple       |    >= 13.x     |
+-------------------+----------------+

Functionality test
------------------

How to know if codejail is working
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The easiest way to test if codejail is working is to validate it in ``Studio`` with a course

Reviewer: ```suggestion
The easiest way to test whether codejail is working is to validate it in ``Studio`` with a course
```

that has loncapa problems exercises.

> This test was performed on the Maple version of Open Edx, using the course ``course_codejail_example.tar.gz``

Reviewer: can you link the course?

Author: Of course

> This test was performed on the Maple version of Open Edx, using the course ``course_codejail_example.tar.gz``

Reviewer: ```suggestion
This test was performed on the Maple version of Open edX, using the course ``course_codejail_example.tar.gz``
```

found in the additional resources section.

> Once the course is imported, enter any section and select an exercise, the result that we must obtain in the

Reviewer: ```suggestion
Once the course is imported, go to any section and select an exercise, the proper result is:
```

Reviewer: any section uses codejail?

Author: All sections use codejail.

example course is the following:

.. image:: ./resourcesfiles/Codejailworking.png
    :width: 725px
    :align: center
    :alt: Example when codejail is working

> Indeed, the content of the course in the section will be presented correctly, without any type of errors and can
> be verified by doing what is found in the instructions of the course.

Reviewer: ```suggestion
In this case, the section's content will render correctly and will be working as specified in the instructions of the problem.
```

Possible failure case
~~~~~~~~~~~~~~~~~~~~~

In case you forget to run ``tutor local init --limit=codejail`` for apparmor profile, the error you can find in 
``Studio`` is as follows::

    Error formatting HTML for problem:
    cannot create LoncapaProblem block-v1:edX+DemoX+Demo_Course+type@problem+block@integral1: Error while
    executing script code: Codejail API Service is unavailable. Please try again in a few minutes.

.. image:: ./resourcesfiles/Codejailfail.png
    :width: 750px
    :align: center
    :alt: Example when codejail is not working

This indicates that the codejail service is not turned on or is not working properly. Be sure to follow the

> steps in Usage so this doesn't happen.

Reviewer: ```suggestion
steps in the usage section so this doesn't happen.
```
And please add the reference to the section
